File: lib/ios/Classes/LdkBroadcaster.swift
//
//  LdkBroardcaster.swift
//  react-native-ldk
//
//  Created by Jason van den Berg on 2022/05/10.
//

import Foundation
import LightningDevKit

class LdkBroadcaster: BroadcasterInterface {    
    override func broadcastTransactions(txs: [[UInt8]]) {
        txs.forEach { tx in
            LdkEventEmitter.shared.send(withEvent: .broadcast_transaction, body: ["tx": Data(tx).hexEncodedString()])
        }
    }
}

File: lib/android/src/main/java/com/reactnativeldk/classes/LdkBroadcaster.kt
package com.reactnativeldk.classes

import com.facebook.react.bridge.Arguments
import com.reactnativeldk.EventTypes
import com.reactnativeldk.LdkEventEmitter
import com.reactnativeldk.hexEncodedString
import org.ldk.structs.BroadcasterInterface

class LdkBroadcaster {
    var broadcaster = BroadcasterInterface.new_impl{ txs ->
        txs.forEach { tx ->
            val body = Arguments.createMap()
            body.putString("tx", tx.hexEncodedString())
            LdkEventEmitter.send(EventTypes.broadcast_transaction, body)
        }
    }
}